### PR TITLE
[W5.11][W10-A1] Gloria Ho Yan Law

### PR DESCRIPTION
**Pull request by glorialaw**
@okkhoy made 2 changes to the program:
1. I added a sort function
2. I made the find command case insensitive

### CODE REVIEW

**Comment by ongkc** [comment]
Good feature implemented, but you can try to use lambdas comparator, a more flexible and elegant way of implementing sorting of objects. Also, remember to update user guide, and I/O testing.

**Comment by okkhoy** [comment]
@glorialaw Please add a description of the enhancement you have made and tag me again so that I can review the work.

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -36,6 +37,12 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
         return String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, personsDisplayed.size());
     }
 
+    /**
+     * Constructs a feedback message to summarise an operation that displayed a sorted list of persons.
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java:41]: You can make this concise!

File: src/seedu/addressbook/commands/FindCommand.java [hunk 2]
```diff
@@ -17,7 +18,7 @@
     public static final String COMMAND_WORD = "find";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
-            + "the specified keywords (case-sensitive) and displays them as a list with index numbers.\n"
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/FindCommand.java:21]: :+1:

File: src/seedu/addressbook/commands/FindCommand.java [hunk 3]
```diff
@@ -40,6 +41,17 @@ public CommandResult execute() {
         return new CommandResult(getMessageForPersonListShownSummary(personsFound), personsFound);
     }
 
+    /**
+     * Converts the Hash Set to upper case
+     * @param convertedSet
+     * @return
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/FindCommand.java:47]: Do not retail empty parameters or return statements in the javadoc comments.

File: src/seedu/addressbook/commands/HelpCommand.java [hunk 4]
```diff
@@ -25,6 +25,8 @@ public CommandResult execute() {
                 + "\n" + ViewAllCommand.MESSAGE_USAGE
                 + "\n" + HelpCommand.MESSAGE_USAGE
                 + "\n" + ExitCommand.MESSAGE_USAGE
+                + "\n" + SortCommand.MESSAGE_USAGE
+                + "\n" + DummyDataCommand.MESSAGE_USAGE
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/HelpCommand.java:29]: `DummyDataCommand`?

File: src/seedu/addressbook/data/person/Person.java [hunk 5]
```diff
@@ -59,6 +60,10 @@ public UniqueTagList getTags() {
         return new UniqueTagList(tags);
     }
 
+    public int compareTo(Person p2) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Person.java:63]: You need to add header comments to this method.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 6]
```diff
@@ -1,11 +1,6 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/UniquePersonList.java:3]: Don't import everything using *; import only the necessary classes. Think why?

**Review by okkhoy — COMMENTED** [review]
You haven't added tests for these enhancements. Please update the PR before the next tutorials to include the tests (either JUnit tests or the IO tests)

**Comment by okkhoy** [comment]
@glorialaw 
Some comments added. Please ack & update the PR after reading comments.

**Comment by glorialaw** [comment]
@okkhoy I updated it to add tests for the find and sort commands (:

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TypicalPersons;
+
+import java.util.*;
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:8]: Don't import everything using *; import only the necessary classes explicitly.

**Review by okkhoy — COMMENTED** [review]
Good job on updating the tests. You can go ahead and close the PR @glorialaw

**Pull request closed without merging**